src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -148,22 +148,25 @@ class MarathonScheduler(
     }
   }
 
-  protected def updateHomeRegion(masterInfo: MasterInfo): Unit = {
+  protected def updateHomeFaultDomain(masterInfo: MasterInfo): Unit = {
     if (masterInfo.hasDomain && masterInfo.getDomain.hasFaultDomain) {
-      homeRegion = Some(masterInfo.getDomain.getFaultDomain.getRegion.getName)
+      homeFaultDomain = Some(FaultDomain(
+        masterInfo.getDomain.getFaultDomain.getRegion.getName,
+        masterInfo.getDomain.getFaultDomain.getZone.getName
```

Comment:
I see where you get the name (mesos)... still don't like it

Reply:
The issue as I see it with fault domain... given a default cluster (3 racks of 3 nodes each).   Dan could have 1 cluster with 3 fault domains (each rack).  and still may want another region to support work load bursting.     What is the "homeFaultDomain"?